### DISCUSSION TITLE
Overlay 4K image onto image and resize

### DISCUSSION THREAD

**Discussion by jwmoss** [ImageMagick/ImageMagick#2082] [Help] [answered]
Hi all!

I'm trying to overlay `4k_logo.jpg` onto `~/Downloads/The\ Avengers\ \(2012\).jpeg` at the top of the banner, like [this](https://movie-posters.herokuapp.com/t1F8xnBHZlQMkSAacfhUUXhmcgo_4k.jpg), but when I do it, the 4k_logo is "zoomed" in on the final image, like [this](https://imgur.com/8KalwyE). 

How do I re-size the 4k_logo to fit the poster? This is what I tried to get this far:

`composite -gravity North ./4k_logo.jpg ~/Downloads/The\ Avengers\ \(2012\).jpeg avengers_4k.jpg`

Version: ImageMagick 7.0.10-15 Q16 x86_64 2020-05-30 https://imagemagick.org, installed via HomeBrew
OS: macOS 10.15.5

**Comment by GeeMack** [comment, marked as answer]
Using ImageMagick v7 you can just use "magick". Then you'd put the target image first followed by the image you're using as the overlay. I'm not as familiar with IM's "composite" utility, but I'd just use "magick" and do a "-composite" operation something like this...

     magick base_img.jpg logo_img.jpg -gravity north -composite output_img.jpg

**Reply by jwmoss** [reply]
Hi @GeeMack 

Thanks for this. Unfortunately, this provides the same output, where the image is overlayed, but the logo is "zoomed in" and not automatically formatted.

**Reply by GeeMack** [reply]
Your example image link in the OP doesn't work. But if you want to resize the logo image to a particular size, or maybe to a percentage of the base image, you can use FX expressions something like this...

    magick base_img.jpg logo_img.jpg -resize x%[fx:t?u.h*0.1:u.h] -gravity north -composite output.jpg

That reads in the base image then the logo, resizes the logo to 10% the height of the base image, then composites it on the destination centered on the north edge.

You may need to put the FX expression in double quotes if your shell isn't happy with any of the [ ] ? % type characters.

**Reply by jwmoss** [reply]
I ran this:

`magick ~/Downloads/The\ Avengers\ \(2012\).jpeg ~/Documents/4k_logo.jpg -resize x%[fx:t?u.h*0.1:u.h] -gravity north -composite ~/Desktop/4k_avengers.jpg`

And got an error:

`no matches found: x%[fx:t?u.h*0.1:u.h]`

I wonder if I'm formatting the cmd incorrectly?

**Reply by GeeMack** [reply]
Looks like you need to put double quotes around that FX expression for your shell.

     ... -resize x"%[fx:t?u.h*0.1:u.h]" ...

**Reply by jwmoss** [reply]
Now we're getting there! It appears that, while this worked, I'm not sure I understand how to dynamically "stretch" the 4k logo to take up the top portion of the image

what the previous command did: [result of previous command](https://imgur.com/a/wXKL2ux)
what I'm trying to do: [existing image I'm trying to re-create](https://movie-posters.herokuapp.com/RYMX2wcKCBAr24UyPD7xwmjaTn_4k.jpg)

**Reply by GeeMack** [reply]
I'd resize that logo image to the desired height with that "-resize" operation, then I'd use "-extent" with some more FX expressions to enlarge the logo's container, with a black background around the logo, to the width of the base image. Then do the composite.

    magick baseimg.jpg logoimg.jpg -resize x"%[fx:t?u.h*0.1:u.h]" \
       -background black -gravity north -extent "%[fx:u.w]x%[fx:s.h]" -composite outimg.jpg

The first FX expression resizes the logo to 0.1 of the height of the base image and leaves the base image untouched. The "-extent" FX expressions stretch the logo's canvas to the width of the base image "%[fx:u.w]" and both images to their own height "%[fx:s.h]".

**Reply by jwmoss** [reply]
Awesome! Thanks so much @GeeMack.
